homeassistant/bootstrap.py

```diff
@@ -325,11 +327,26 @@ async def _async_set_up_integrations(
 
     async def async_setup_multi_components(domains: Set[str]) -> None:
         """Set up multiple domains. Log on failure."""
+
+        async def _async_log_pending_setups() -> None:
+            """Periodic log of setups that are pending for longer than LOG_SLOW_STARTUP_INTERVAL."""
+            while True:
+                await asyncio.sleep(LOG_SLOW_STARTUP_INTERVAL)
+                remaining = [
```

Comment:
This is not ideal. Now it measures also when we wait for installing requirements. 

Ideally we would wrap the setup with a:

```python
hass.data[DATA_SETUP_STARTED][domain] = dt_util.utcnow()
try:
    if sync:
        task = hass.async_add_executor_job(…)
    elif async:
        task = async_setup(…)
finally:
    hass.data[DATA_SETUP_STARTED].pop(domain)
```

Reply:
Makes sense.  Will rework when I've had more sleep.

Reply:
Adjusted